packages/accounts/src/msca/plugins/session-key/signer.ts

```diff
@@ -0,0 +1,131 @@
+import {
+  LocalAccountSigner,
+  SignerSchema,
+  type SignTypedDataParams,
+  type SmartAccountSigner,
+} from "@alchemy/aa-core";
+import type { Hex, PrivateKeyAccount } from "viem";
+import { generatePrivateKey } from "viem/accounts";
+import { z } from "zod";
+
+export const createSessionKeySignerSchema = <
+  TFallback extends SmartAccountSigner
+>() => {
+  return z.object({
+    storageType: z
+      .union([z.literal("local-storage"), z.literal("session-storage")])
+      .default("local-storage"),
+    storageKey: z.string().default("session-key-signer:session-key"),
+    fallbackSigner: z.custom<TFallback>((signer) => SignerSchema.parse(signer)),
+  });
+};
+
+export type SessionKeySignerConfig<TFallback extends SmartAccountSigner> =
+  z.input<ReturnType<typeof createSessionKeySignerSchema<TFallback>>>;
+
+/**
+ * A simple session key signer that uses localStorage or sessionStorage to store
+ * a private key. If the key is not found, it will generate a new one and store
+ * it in the storage.
+ */
+export class SessionKeySigner<TFallback extends SmartAccountSigner>
+  implements SmartAccountSigner<LocalAccountSigner<PrivateKeyAccount>>
+{
+  signerType: string = "alchemy:session-key";
```

Comment:
might want to append the fallback signer's signer type for logging

Reply:
oh true